judge/views/problem.py

```diff
@@ -213,8 +226,117 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.can_vote(user, self.object)  # if this problem is votable by this user
+        # the vote this user has already cast on this problem
+        if context['can_vote']:
+            vote = ProblemPointsVote.objects.filter(voter=user.profile, problem=self.object)
+        # whether or not they've already voted
+        context['has_voted'] = context['can_vote'] and vote.exists()
+        if context['has_voted']:
+            context['voted_points'] = vote.first().points  # the previous vote's points
+            context['voted_note'] = vote.first().note
+
+        if 'problem_points_vote_form' not in context:
+            context['problem_points_vote_form'] = ProblemPointsVoteForm({})
+
+        if 'has_errors' not in context:
+            context['has_errors'] = False
+
+        if 'points_placeholder' not in context:  # placeholder for the points
+            if context['has_voted']:  # if voted, the vote
+                context['points_placeholder'] = context['voted_points']
+            else:  # otherwise a *nice* default
+                context['points_placeholder'] = 69.42069
+
+        if 'note_placeholder' not in context:
+            if context['has_voted']:
+                context['note_placeholder'] = context['voted_note']
+            else:
+                context['note_placeholder'] = self.default_note()
+
+        all_votes = sorted([v.points for v in ProblemPointsVote.objects.filter(problem=self.object)])
+        context['has_votes'] = len(all_votes) > 0
+        if context['has_votes']:
+            context['mean_vote'] = sum(all_votes) / len(all_votes)
+            context['num_votes'] = len(all_votes)
+            context['min_vote'] = all_votes[0]
+            context['max_vote'] = all_votes[-1]
+
+            # provides index and value of the median of some range of the data
+            def median(left_index, right_index, data):
+                size = right_index - left_index + 1
+                index = left_index + (size - 1) / 2
+                value = None
```

Comment:
The assignment to None is actually not necessary in Python. In addition, is there any need for the if statement below? It seems taking the average of the floor and ceil indices should work for both even and odd.